lionweb/src/main/kotlin/com/strumenta/kolasu/lionweb/StarLasuLWLanguage.kt

```diff
@@ -21,6 +21,29 @@ object StarLasuLWLanguage : Language("com.strumenta.StarLasu") {
         }
         astNode.addReference("originalNode", astNode, Multiplicity.OPTIONAL)
         astNode.addReference("transpiledNode", astNode, Multiplicity.MANY)
+
+        addPlaceholderNodeAnnotation(astNode)
+    }
+
+    private fun addPlaceholderNodeAnnotation(astNode: Concept) {
+        val placeholderNodeAnnotation = Annotation(
+            this,
+            PLACEHOLDER_NODE,
+            idForContainedElement(PLACEHOLDER_NODE),
+            keyForContainedElement(PLACEHOLDER_NODE)
+        )
+        placeholderNodeAnnotation.annotates = LionCore.getConcept()
+        val reference =
+            Reference().apply {
+                this.name = "originalNode"
+                this.id = "${placeholderNodeAnnotation.id!!.removeSuffix("-id")}-$name-id"
+                this.key = "${placeholderNodeAnnotation.key!!.removeSuffix("-key")}-$name-key"
+                this.type = astNode
+                this.setOptional(true)
```

Comment:
I think it should not be optional because if the annotation is present, it should always refer to exactly one node that this is acting as a placeholder for

Reply:
It's optional because:
1. the origin is optional in Kolasu as well
2. the origin could be something that cannot be represented in Lionweb, such as a parse tree node